scalalib/worker/src/ZincWorkerImpl.scala

```diff
@@ -330,10 +333,14 @@ class ZincWorkerImpl(compilerBridge: Either[
           grepJar(compilerClasspath, s"dotty-compiler_${scalaBinaryVersion(scalaVersion)}", scalaVersion)
         else
           compilerJarNameGrep(compilerClasspath, scalaVersion)
+
       val scalaInstance = new ScalaInstance(
         version = scalaVersion,
         loader = getCachedClassLoader(compilersSig, combinedCompilerJars),
-        libraryJar = libraryJarNameGrep(compilerClasspath, scalaVersion).toIO,
+        libraryJar = libraryJarNameGrep(
+          compilerClasspath,
+          if (isDotty(scalaVersion)) "2.13.0" else scalaVersion
```

Comment:
:no_entry:  This is going to do the wrong thing with more recent versions of dotty which depend on 2.13.1, and will break again when we switch to 2.13.2, etc. The root problem here is that you have this scala-library-0.xx jar at all, you shouldn't be using it or depending on it in any way. You should depend on dotty-library which depends on the appropriate scala-library-2.13 and that's all.

Reply:
@smarter Thanks for reviewing it!

@lihaoyi Are you going to do something about it? If not, we might just open a ticket?

Reply:
Perhaps just open a ticket for now? Then if someone wants to fix it to support a new version of Dotty, and adjust/add to the test suite, they can submit a PR to do so.

@smarter why does the minor version of scala-library matter for Dotty? Isn't scala-library meant to be backwards and forwards binary compatible between minor versions?

Reply:
>  why does the minor version of scala-library matter for Dotty?

You're calling `libraryJarNameGrep` which is going to look for a jar called scala-library-2.13.0 among the dependencies of the dotty compiler, since recent versions of the dotty compiler depend on scala-library-2.13.1, it won't find it. (and we do have to bump the version of scala-library we depend on sometimes, e.g. for https://github.com/lampepfl/dotty/pull/8480 we'll need to depend on 2.13.2)